lib/awful/widget/layoutlist.lua

```diff
@@ -0,0 +1,442 @@
+----------------------------------------------------------------------------
+--- Display the available client layouts for a screen.
+--
+-- This is what the layoutlist looks like by default with a vertical layout:
+--
+--@DOC_awful_widget_layoutlist_default_EXAMPLE@
+--
+-- In the second example, it is shown how to create a popup in centered in
+-- the screen:
+--
+--@DOC_awful_widget_layoutlist_popup_EXAMPLE@
+--
+-- This example show how to extend the `awful.widget.layoutbox` to show a layout
+-- list popup:
+--
+--@DOC_awful_widget_layoutlist_bar_EXAMPLE@
+--
+-- This example show how to add a layout subset to the default wibar:
+--
+--@DOC_awful_widget_layoutlist_wibar_EXAMPLE@
+--
+-- @author Emmanuel Lepage Vallee &lt;elv1313@gmail.com&gt;
+-- @copyright 2010, 2018 Emmanuel Lepage Vallee
+-- @classmod awful.widget.layoutlist
+----------------------------------------------------------------------------
+
+local capi     = {screen = screen, tag = tag}
+local wibox    = require("wibox")
+local awcommon = require("awful.widget.common")
+local abutton  = require("awful.button")
+local ascreen  = require("awful.screen")
+local gtable   = require("gears.table")
+local beautiful= require("beautiful")
+local alayout  = require("awful.layout")
+local surface  = require("gears.surface")
+
+local module = {}
+
+local default_buttons = gtable.join(
```

Comment:
I recall seeing a PR removing the joins from `rc.lua`. Maybe not needed here, too?

Reply:
That pull request isn't merge, so it's still needed for now